cli/azd/cmd/templates.go

```diff
@@ -316,9 +317,18 @@ func (a *templateSourceListAction) Run(ctx context.Context) (*actions.ActionResu
 
 func newTemplateSourceAddCmd() *cobra.Command {
 	return &cobra.Command{
-		Use:   "add <key>",
-		Short: fmt.Sprintf("Adds an azd template source at the specified key %s", output.WithWarningFormat("(Beta)")),
-		Args:  cobra.ExactArgs(1),
+		Use: "add <key>",
+		Short: fmt.Sprintf(
+			"Adds an azd template source at the provided key %s\n"+
+				"  Use known-keys w/o a type for:\n"+
+				"   ・default: azd built-in template list\n"+
+				"   ・awesome-azd: Includes all templates from http://aka.ms/awesome-azd\n"+
+				"  For custom key, use type and location flags. Supported types:\n"+
+				"   ・file: local file only \n"+
+				"   ・url: public urls, only, to download json source\n"+
+				"   ・gh: path to json file in a GitHub repository (supports private repos)",
```

Comment:
Part of me wishes the way this worked was that we just had a `github://` support for the `url` scheme (and have `file` be syntatic sugar for a `url` type with the `file://` scheme.  In that world I would only support the following URL format `github://<host>/<owner>/<repo>/<branch>/<path>` (if host is empty, public github is assumed but this would allow private instances of GH enterprise, I think) and then basically translate this into calls to calls against the API but auth'd but the current instance of `gh` on the box?

Reply:
I feel like asking for file scheme brings an extra step for customers to create the expected scheme.

I'm trying to make it super simple for customers to copy/paste the url from the browser when they are on github website. Either if you are just browsing the file or you click the `raw` button, you can then just copy the url into azd.

Also, the url scheme would be yet another type after the already defined `--type` flag. Which feels like mixing two input strategies (flag + naming-convention).  If we prefer naming convention, I would expect no `--type` flag at all, as we can use the convention to just do `azd template source add [customeKey | known-key] [github://, file://, https://, etc]`